lib/extensions/attachable/active_record/base.rb

```diff
@@ -160,32 +159,55 @@ def attachment_reference
     def attachment_reference=(attachment_reference)
       return self.attachment_reference if self.attachment_reference == attachment_reference
 
-      attribute_will_change!(ATTACHMENT_ATTRIBUTE)
+      mark_attachment_as_changed(self.attachment_reference)
       attachment_references.clear
       attachment_references << attachment_reference if attachment_reference
     end
     alias_method :attachment=, :attachment_reference=
 
     def build_attachment_reference(attributes = {})
-      attribute_will_change!(ATTACHMENT_ATTRIBUTE)
+      mark_attachment_as_changed(attachment_reference)
       attachment_references.clear
       attachment_references.build(attributes)
     end
     alias_method :build_attachment, :build_attachment_reference
 
-    def attachment_reference_changed?
-      changed.include?(ATTACHMENT_ATTRIBUTE)
-    end
-    alias_method :attachment_changed?, :attachment_reference_changed?
-
     def file=(file)
       if file
         build_attachment_reference(file: file)
       else
         return nil if attachment_reference.nil?
-        attribute_will_change!(ATTACHMENT_ATTRIBUTE)
+        mark_attachment_as_changed(attachment_reference)
         attachment_references.clear
       end
     end
+
+    # Tracks the the attachment_reference changes and support clear/revert the changes.
+    # `attachment_reference` is a virtual attribute, and changes tracking to such attributes are not well supported in
+    # rails 5: https://github.com/rails/rails/issues/25787
+    def attachment_reference_changed?
```

Comment:
Once we upgrade to Rails 5.1 we should revert this commit?

Reply:
Possible, this commit gets merged in 5.1 which could support that (again): https://github.com/marketplacer/rails/commit/45e098898aa92189dfe98b93aee597602c7fc322 

But have to double check that time.